src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpoint.java

```diff
@@ -194,14 +197,26 @@ public KeyMaterialFactory newKeyMaterialFactory(@Nonnull AbstractBuild build) th
     }
 
     /**
+     * @deprecated Call {@link #newKeyMaterialFactory(Item, VirtualChannel, Launcher, TaskListener)}
+     */
+    @Deprecated
+    public KeyMaterialFactory newKeyMaterialFactory(Item context, @Nonnull VirtualChannel target) throws IOException, InterruptedException {
+        return newKeyMaterialFactory(context, target, null, TaskListener.NULL);
+    }
+
+    /**
      * Makes the credentials available locally and returns {@link KeyMaterialFactory} that gives you the parameters
      * needed to access it.
      */
-    public KeyMaterialFactory newKeyMaterialFactory(Item context, @Nonnull VirtualChannel target) throws IOException, InterruptedException {
+    public KeyMaterialFactory newKeyMaterialFactory(@CheckForNull Item context, @Nonnull VirtualChannel target, @CheckForNull Launcher launcher, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        if (credentialsId == null) {
+            return KeyMaterialFactory.NULL; // nothing needed to be done
+        }
         DockerRegistryToken token = getToken(context);
-        if (token==null)    return KeyMaterialFactory.NULL;    // nothing needed to be done
-
-        return token.newKeyMaterialFactory(getEffectiveUrl(), target);
+        if (token == null) {
+            throw new AbortException("Could not find credentials matching " + credentialsId);
```

Comment:
Previously it would silently succeed, allowing subsequent attempts to use the registry to fail mysteriously.